packages/scene-composer/src/components/three-fiber/EntityGroup/index.tsx

```diff
@@ -22,12 +23,35 @@ interface IEntityGroupProps {
   node: ISceneNodeInternal;
 }
 
+const subModelChildrenDisabled = !getGlobalSettings().featureConfig[COMPOSER_FEATURES.SubModelChildren];
+
 const ChildGroup = ({ node }: { node: ISceneNodeInternal }) => {
   const sceneComposerId = useSceneComposerId();
   const { getSceneNodeByRef } = useSceneDocument(sceneComposerId);
 
+  const getAllChildrenRefs = (node: ISceneNodeInternal) => {
+    const childRefs: string[] = [];
+    if (node.childRefs) {
+      childRefs.push(...node.childRefs);
+
+      if (subModelChildrenDisabled) {
+        node.childRefs.forEach((childRef) => {
+          const child = getSceneNodeByRef(childRef);
+          const childIsSubModel = !!findComponentByType(child, KnownComponentType.SubModelRef);
+          if (child && childIsSubModel) {
+            childRefs.push(...child.childRefs);
```

Comment:
You only handled 1 level of nesting, can there be multiple levels of nestings? e.g. a GLTF model contains another GLTF model and both having submodels, would this code work?

Reply:
This is a good call out and I almost did that. However by design sub models will not be able to have sub model children. And because of that as long as they become a child of something in real space a.k.a whatever its parent is by definition we are safe